### PR TITLE
Fix some LGTM errors of "Multiplication result converted to larger type"

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR fixes some of the errors on [Godot's LGTM page](https://lgtm.com/projects/g/godotengine/godot/alerts?mode=list).

### CODE REVIEW

File: servers/audio/audio_rb_resampler.cpp [hunk 1]
```diff
@@ -176,8 +176,8 @@ Error AudioRBResampler::setup(int p_channels, int p_src_mix_rate, int p_target_m
 		rb_bits = desired_rb_bits;
 		rb_len = (1 << rb_bits);
 		rb_mask = rb_len - 1;
-		rb = memnew_arr(float, rb_len *p_channels);
-		read_buf = memnew_arr(float, rb_len *p_channels);
+		rb = memnew_arr(float, rb_len *(size_t)p_channels);
+		read_buf = memnew_arr(float, rb_len *(size_t)p_channels);
```
Comment (aaronfranke) [inline servers/audio/audio_rb_resampler.cpp:180]: This code is a bit weird because this is definitely a multiplication, but clang-format insists that there shouldn't be a space after the asterisk (does it think this has something to do with pointers?)

File: scene/2d/parallax_layer.cpp [hunk 2]
```diff
@@ -123,12 +123,12 @@ void ParallaxLayer::set_base_offset_and_scale(const Point2 &p_offset, real_t p_s
...
 
 	if (mirroring.x) {
-		double den = mirroring.x * p_scale;
+		real_t den = mirroring.x * p_scale;
 		new_ofs.x -= den * ceil(new_ofs.x / den);
 	}
 
 	if (mirroring.y) {
-		double den = mirroring.y * p_scale;
+		real_t den = mirroring.y * p_scale;
```
Comment (ellenhp) [inline scene/2d/parallax_layer.cpp:131]: Why is this necessary? Not trying to second-guess you, just wondering because I've never seen real_t used before.

File: servers/audio/audio_rb_resampler.cpp (hunk 1 above)
Comment (ellenhp) [reply servers/audio/audio_rb_resampler.cpp:180]: Could you try reversing the order of the operands, see if clang-format likes that better? I know it was like this before too, but it would read a bit more cleanly with a space in there.

**Review by ellenhp — COMMENTED** [review]
Looks good overall. Just a question and a suggestion for something to try to fix the clang-format weirdness.

File: scene/2d/parallax_layer.cpp (hunk 2 above)
Comment (aaronfranke) [reply scene/2d/parallax_layer.cpp:131]: `real_t` is the abstract type that Godot uses for vectors and much more. It can be defined as either float or double. In the audio code you'll see `float` because you don't need more precision than that when working with audio.

File: servers/audio/audio_rb_resampler.cpp (hunk 1 above)
Comment (aaronfranke) [reply servers/audio/audio_rb_resampler.cpp:180]: I swapped them around and clang-format still removes the space.

Comment (akien-mga) [reply servers/audio/audio_rb_resampler.cpp:180]: Since we're calculating it twice, maybe do `const size_t arr_size = rb_len * p_channels;` beforehand.

There's also this one where it's bogus:
```
platform/javascript/audio_driver_javascript.cpp
122:    output_rb = memnew_arr(float, buffer_length *channel_count);
129:    input_rb = memnew_arr(float, buffer_length *channel_count);
```

Comment (ellenhp) [reply servers/audio/audio_rb_resampler.cpp:180]: I don't think we should block this PR on fixing that issue but it does bother me.

Comment (aaronfranke) [reply servers/audio/audio_rb_resampler.cpp]: @ellenhp Nah, it's an easy fix on the same lines, so it makes sense to do it in this PR instead of waiting.

I've updated this PR to have these values be constants.

Comment (ellenhp) [reply servers/audio/audio_rb_resampler.cpp]: Oh, I just meant the clang-format issue when I said that.

Comment (aaronfranke) [reply servers/audio/audio_rb_resampler.cpp]: The formatting has to be correct or else the CI checks will fail. The work-around of adding constants is a simple solution.

If you're talking about fixing the upstream bug in clang-format, yes I agree, that would take a very long time for the fix to arrive downstream even assuming we were able to get it fixed quickly so we shouldn't wait for that.

**Review by akien-mga — APPROVED** [review]
